client/src/CodeClaimPage/ClaimForm.tsx

```diff
@@ -12,44 +12,65 @@ import { hasWeb3 } from '../poap-eth';
 import { SubmitButton } from '../components/SubmitButton';
 import ClaimFooterMessage from './ClaimFooterMessage';
 
-/* Assets */
+// lib
+import { COLORS } from '../lib/constants';
+
+// types
+import { TemplatePageFormValues } from '../templates/TemplateFormPage/components/TemplateForm';
+import { useImageSrc } from '../lib/hooks/useImageSrc';
 
 type QRFormValues = {
   address: string;
 };
 
 const ClaimForm: React.FC<{
-  claim: HashClaim;
+  claim?: HashClaim;
   method: string;
+  template?: TemplatePageFormValues;
   onSubmit: (claim: HashClaim) => void;
-}> = ({ claim, onSubmit, method }) => {
+}> = ({ claim, onSubmit, method, template }) => {
   const [enabledWeb3, setEnabledWeb3] = useState<boolean | null>(null);
   const [account, setAccount] = useState<string>('');
 
+  const mobileImageUrlRaw = claim?.event_template?.mobile_image_url
+    ? claim?.event_template?.mobile_image_url
+    : template?.mobile_image_url;
+  const mobileImageLink = claim?.event_template?.mobile_image_link
+    ? claim?.event_template?.mobile_image_link
+    : template?.mobile_image_link;
+  const mainColor = claim?.event_template?.main_color
+    ? claim?.event_template?.main_color
+    : template?.main_color;
```

Comment:
We can use `nullish coalescing` operator here to avoid ternary operator

```suggestion
  const mobileImageUrlRaw = claim?.event_template?.mobile_image_url ??template?.mobile_image_url;
  const mobileImageLink = claim?.event_template?.mobile_image_link ??template?.mobile_image_link;
  const mainColor = claim?.event_template?.main_color ?? template?.main_color;
```